src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/MergeAssetManifests.cs

```diff
@@ -67,10 +67,10 @@ public override bool Execute()
 
             foreach (var assetManifestXml in assetManifestXmls)
             {
-                packageElements.AddRange(assetManifestXml.Descendants("Package"));
-                blobElements.AddRange(assetManifestXml.Descendants("Blob"));
+                packageElements.AddRange(assetManifestXml.Descendants("Package").Where(package => package.Attribute("Visibility")?.Value != "Vertical"));
```

Comment:
Consider including filters for the other visibilities in case they happened to make it through. Or, perhaps a 
check+exception since we don't expect anything but vertical and external here.

Reply:
`Internal` would also be valid here (this runs as the end of a single vertical, so we'd still see and want to upload `Internal` visibility assets)

Reply:
I believe we need the `Internal` assets to flow through the internal manifests so our logic to populate Versions.props from the input manifests (including from previous build passes) would find the packages.